File: project/app/src/main/java/com/iven/musicplayergo/adapters/ColorsAdapter.java
package com.iven.musicplayergo.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.iven.musicplayergo.R;

import java.util.Random;

public class ColorsAdapter extends RecyclerView.Adapter<ColorsAdapter.SimpleViewHolder> {

    //fixed int array of accent colors
    private static final int[] colors = new int[]{
            R.color.red,
            R.color.pink,
            R.color.purple,
            R.color.deep_purple,
            R.color.indigo,
            R.color.blue,
            R.color.light_blue,
            R.color.cyan,
            R.color.teal,
            R.color.green,
            R.color.amber,
            R.color.orange,
            R.color.deep_orange,
            R.color.brown,
            R.color.gray,
            R.color.blue_gray
    };
    private final Context mContext;
    private final int mAccent;
    private final AccentChangedListener mOnAccentChangedListener;

    public ColorsAdapter(@NonNull final Context context, final int accent) {
        mContext = context;
        mAccent = accent;
        mOnAccentChangedListener = (AccentChangedListener) mContext;
    }

    public static int getRandomColor() {
        final int rnd = new Random().nextInt(colors.length);
        return colors[rnd];
    }

    @Override
    @NonNull
    public SimpleViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {

        final View itemView = LayoutInflater.from(mContext)
                .inflate(R.layout.color_option, parent, false);

        return new SimpleViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull final SimpleViewHolder holder, final int position) {

        final int color = colors[holder.getAdapterPosition()];
        final int drawable = color != mAccent ? R.drawable.ic_checkbox_blank : R.drawable.ic_checkbox_marked;
        final int parsedColor = ContextCompat.getColor(mContext, color);
        holder.color.setImageResource(drawable);
        holder.color.setColorFilter(parsedColor);
    }

    @Override
    public int getItemCount() {

        return colors.length;
    }

    public interface AccentChangedListener {
        void onAccentChanged(final int color);
    }

    class SimpleViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private final ImageView color;

        SimpleViewHolder(@NonNull final View itemView) {
            super(itemView);

            color = (ImageView) itemView;
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(@NonNull final View v) {
            //recreate the activity only if necessary
            final int color = colors[getAdapterPosition()];
            if (color != mAccent) {
                mOnAccentChangedListener.onAccentChanged(color);
            }
        }
    }
}

File: project/app/src/main/java/com/iven/musicplayergo/adapters/ColorsAdapter.kt
package com.iven.musicplayergo.adapters

import android.app.Activity
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.iven.musicplayergo.R
import com.iven.musicplayergo.Utils
import com.iven.musicplayergo.mMusicPlayerGoPreferences

class ColorsAdapter(private val activity: Activity) :
    RecyclerView.Adapter<ColorsAdapter.ColorsHolder>() {

    //fixed int array of accent colors
    private val mColors = intArrayOf(
        R.color.red,
        R.color.pink,
        R.color.purple,
        R.color.deep_purple,
        R.color.indigo,
        R.color.blue,
        R.color.light_blue,
        R.color.cyan,
        R.color.teal,
        R.color.green,
        R.color.amber,
        R.color.orange,
        R.color.deep_orange,
        R.color.brown,
        R.color.gray,
        R.color.blue_gray
    )

    private var mSelectedColor = R.color.blue

    init {
        mSelectedColor = mMusicPlayerGoPreferences.accent
    }

    private fun getColorPosition(color: Int): Int {
        return try {
            mColors.indexOf(color)
        } catch (e: Exception) {
            0
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ColorsHolder {
        return ColorsHolder(
            LayoutInflater.from(parent.context).inflate(
                R.layout.color_option,
                parent,
                false
            )
        )
    }

    override fun getItemCount(): Int {
        return mColors.size
    }

    override fun onBindViewHolder(holder: ColorsHolder, position: Int) {
        holder.bindItems(mColors[holder.adapterPosition])
    }

    inner class ColorsHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bindItems(color: Int) {
            val colorOption = itemView as ImageView
            val drawable = if (color != mSelectedColor) R.drawable.ic_checkbox_blank
            else
                R.drawable.ic_checkbox_marked

            val colorFromInt = Utils.getColor(activity, color, R.color.blue)
            colorOption.setImageResource(drawable)
            colorOption.setColorFilter(colorFromInt)
            itemView.setOnClickListener {

                if (mColors[adapterPosition] != mSelectedColor) {
                    notifyItemChanged(getColorPosition(mSelectedColor))
                    mSelectedColor = mColors[adapterPosition]
                    colorOption.setImageResource(R.drawable.ic_checkbox_marked)
                    mMusicPlayerGoPreferences.accent = mSelectedColor

                    Handler().postDelayed({
                        Utils.applyNewThemeSmoothly(activity)
                    }, 250)
                }
            }
        }
    }
}
